schema/table.go

```diff
@@ -167,6 +179,36 @@ func (tt Tables) ValidateColumnNames() error {
 	return nil
 }
 
+// this will filter the tree in-place
+func (t *Table) filterDfs(tables, skipTables []string) *Table {
+	var matched bool
+	for _, includeTable := range tables {
+		if glob.Glob(includeTable, t.Name) {
+			matched = true
+			break
+		}
+	}
+	for _, skipTable := range skipTables {
+		if glob.Glob(skipTable, t.Name) {
+			matched = false
+			break
+		}
+	}
+	filteredRelations := make([]*Table, 0, len(t.Relations))
+	for _, r := range t.Relations {
+		filteredChild := r.filterDfs(tables, skipTables)
+		if filteredChild != nil {
+			matched = true
```

Comment:
Won't this always return the parent table if the child matches, for example:
```go
               {
			name:                    "should skip parent table",
			tables:                  []*Table{{Name: "main_table", Relations: []*Table{{Name: "sub_table", Parent: &Table{Name: "main_table"}}}}},
			configurationTables:     []string{"*"},
			configurationSkipTables: []string{"main_table"},
			want:                    []string{""},
		},
```

The above test fails as it returns `main_table` and `sub_table`

Reply:
Good catch. it depends on what we define on what has preference. if skip has preference then yes. I've added that logic.

Reply:
Ah I think https://github.com/cloudquery/plugin-sdk/pull/488/files#diff-7e84c216b71c17b23bc66ee33042fe73fc9352c152dbc4fb4cd9a94d74e2bfb4R49 is the reason